### PR TITLE
Update ComfyUI

### PR DESCRIPTION
**Pull request by PassiveLemon**
Just updated ComfyUI to Pytorch 2, Xformers 0.0.20, and the latest Comfy commit. I updated the Docker-compose accordingly.
This also fixes an issue where the mount for the output is not loaded so the container will not save images to a host mounted directory.

I found zero change with and without Triton so I just removed it.

### CODE REVIEW

**Comment by AbdBarho** [comment]
> This also fixes an issue where the mount for the output is not loaded so the container will not save images to a host mounted directory.

hmmm, weird, I did not have this problem.

> I found zero change with and without Triton so I just removed it.

triton now is now automatically installed with pytorch 2.

**Comment by PassiveLemon** [comment]
> > This also fixes an issue where the mount for the output is not loaded so the container will not save images to a host mounted directory.
> 
> hmmm, weird, I did not have this problem.
If you use a "save image" node, it automatically saves to the output folder? It does not for me. It will only work if I specifically add that docker mount.

**Comment by PassiveLemon** [comment]
This is mostly temporary but I pushed it so you can get the idea. I am now currently working on making custom nodes much easier to install.
If there is any directory in the custom_nodes path, it will get searched for a requirements.txt and then be installed. Sometimes they require distro specific packages like cmake so those might also need to get installed in this stage but I don't currently have a way to automatically fetch those which is the main issue here.
Currently, I just put Debian's build-essential package there because it contains a lot of the things nodes might need but extra stuff can always just be added to that line. Still thinking of different solutions so maybe you might have an idea that you like. A separate file for only distro packages could be added for easy user interaction but I'd like to avoid mostly empty stray files. They could also be loaded and installed in the Dockerfile using a similar method but again, its another stray file.

File: services/comfy/entrypoint.sh [hunk 1]
```diff
@@ -44,4 +44,16 @@ for to_path in "${!MOUNTS[@]}"; do
   echo Mounted $(basename "${from_path}")
 done
 
+if [ "$(ls -A /stable-diffusion/custom_nodes)" ]; then
+  chmod 777 -R "/stable-diffusion/custom_nodes/"
+  apt-get install build-essential -y
```
Comment (AbdBarho) [inline services/comfy/entrypoint.sh:49]: we could move this into the dockerfile, where we install aria2 and others

File: services/comfy/entrypoint.sh [hunk 2]
```diff
@@ -44,4 +44,16 @@ for to_path in "${!MOUNTS[@]}"; do
   echo Mounted $(basename "${from_path}")
 done
 
+if [ "$(ls -A /stable-diffusion/custom_nodes)" ]; then
+  chmod 777 -R "/stable-diffusion/custom_nodes/"
```
Comment (AbdBarho) [inline services/comfy/entrypoint.sh:48]: why is this necessary?

File: services/comfy/entrypoint.sh [hunk 3]
```diff
@@ -44,4 +44,16 @@ for to_path in "${!MOUNTS[@]}"; do
...
+  apt-get install build-essential -y
+  for dir in "/stable-diffusion/custom_nodes/*"; do
+    if [ -e "$dir/requirements.txt" ]; then
+      echo $dir
+      cd $dir
+      pip install -r requirements.txt
+    fi
+  done
```
Comment (AbdBarho) [inline services/comfy/entrypoint.sh:56]: I think we could simplify this loop
```sh
shopt -s nullglob
list=(${ROOT}/custom_nodes/*/requirements.txt)
for req in "${list[@]}"; do
  pip install -r "$req"
done
```

**Comment by AbdBarho** [comment]
Yeah extensions are weird, same situation is for `auto`. I had to specifically create a `startup.sh` script so users can put all of their stuff there, usually installing dependencies.

That being said, pip is smart enough to not re-install everything, only missing stuff.

Do you think a `startup.sh` script is a good solution?

File: docker-compose.yml [hunk 4]
```diff
@@ -74,8 +74,10 @@ services:
     <<: *base_service
     profiles: ["comfy"]
     build: ./services/comfy/
-    image: sd-comfy:1
+    image: sd-comfy:2
     tty: true
+    volumes:
+      - *v2
```
Comment (AbdBarho) [inline docker-compose.yml:80]: This is already mounted in the `x-base-service`, so there is no need to change the mounts here.

File: services/comfy/entrypoint.sh (hunk 2 above)
Comment (PassiveLemon) [reply services/comfy/entrypoint.sh:48]: Testing reasons. I wasn't going to bother fixing everything until there was a proposed solution

File: docker-compose.yml (hunk 4 above)
Comment (PassiveLemon) [reply docker-compose.yml:80]: That's the issue though. I know it's present, but the save image node won't save the images to the hosts output directory unless it is present.

**Comment by PassiveLemon** [comment]
> Yeah extensions are weird, same situation is for `auto`. I had to specifically create a `startup.sh` script so users can put all of their stuff there, usually installing dependencies.
> 
> That being said, pip is smart enough to not re-install everything, only missing stuff.
> 
> Do you think a `startup.sh` script is a good solution?

Maybe. It would definitely work, but it would really only be used for the Debian packages since the node pip dependencies would be included in their requirements.txt files. I can whip up a design if you would like

File: docker-compose.yml (hunk 4 above)
Comment (AbdBarho) [reply docker-compose.yml:80]: What is the output folder of the save image node? and does the `output/comfy` exist on your machine?

File: services/comfy/entrypoint.sh [hunk 5]
```diff
@@ -44,4 +44,16 @@ for to_path in "${!MOUNTS[@]}"; do
...
+  . /data/config/comfy/startup.sh
+  popd
+fi
+
+shopt -s nullglob
+list=(${ROOT}/custom_nodes/*/requirements.txt)
+for req in "${list[@]}"; do
+  pip install -r "$req"
+done
+
```
Comment (AbdBarho) [inline services/comfy/entrypoint.sh:58]: I do still think this should be separate, otherwise it could introduce so many bugs we cannot account for.
i.e. we build the container with certain dependencies, yet at startup they get overwritten by random ones from the extensions.

An example which is very frustrating that comes to mind is `open-cv-headless`.

Comment (PassiveLemon) [reply services/comfy/entrypoint.sh:58]: /output/comfy does exist on my system, yes.

How would you want to separate out that part?

**Comment by AbdBarho** [comment]
hey I am going to take the liberty and change some stuff do we can get this merged. stuff that is left out can still be added in a new PR.
